sql/core/src/test/scala/org/apache/spark/sql/CollationStringExpressionsSuite.scala

```diff
@@ -425,6 +425,54 @@ class CollationStringExpressionsSuite
     })
   }
 
+  test("Support Left/Right/Substr with collation") {
+    case class SubstringTestCase(
+        method: String,
+        str: String,
+        len: String,
+        pad: Option[String],
+        collation: String,
+        result: Row) {
+      val strString = if (str == "null") "null" else s"'$str'"
+      val query =
+        s"SELECT $method(collate($strString, '$collation'), $len${pad.map(p => s", '$p'").getOrElse("")})" // scalastyle:ignore line.size.limit
```

Comment:
it's already to break a row

we shouldn't normally use // scalastyle:ignore line.size.limit

Reply:
Done